src/widgetastic_patternfly4/ouia.py

```diff
@@ -0,0 +1,45 @@
+import importlib
+import sys
+
+from widgetastic.utils import ParametrizedLocator
+
+
+class OUIAMixin:
+
+    ROOT = ParametrizedLocator(
+        './/*[@data-ouia-component-type="PF4/{@component_type}" and '
+        '@data-ouia-component-id="{@component_id}"]'
```

Comment:
how do we refer to widgets that don't have a popper component id yet? (or don't need one)

Reply:
Then non-OUIA widget version should be used. According to the OUIA spec:

> if there are multiple instances of a component on the page `data-ouia-component-id` MUST be used